rusoto/signature/src/signature.rs

```diff
@@ -153,14 +155,15 @@ impl SignedRequest {
     ///
     /// Has no effect if the payload is not set, or is not a buffer. Will not
     /// override an existing value for the `Content-MD5` header.
-    pub fn maybe_set_content_md5_header(&mut self) {
+    pub fn maybe_set_content_md5_header(&mut self) -> Result<(), Error> {
         if self.headers.contains_key("Content-MD5") {
-            return;
+            return Ok(());
         }
         if let Some(SignedRequestPayload::Buffer(ref payload)) = self.payload {
-            let digest = Md5::digest(payload);
+            let digest = hash(MessageDigest::md5(), payload)?;
```

Comment:
When does the function get called? I'm bit worried about md5 availability. OpenSSL on most Linux distribution bans md5 by default.

```shell
[root@590025fddcf8 /]# openssl version
OpenSSL 3.0.7 1 Nov 2022 (Library: OpenSSL 3.0.7 1 Nov 2022)
[root@590025fddcf8 /]# openssl md5 /etc/rocky-release
Error setting digest
403CBAAF5D7F0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:373:Global default library context, Algorithm (MD5 : 97), Properties ()
403CBAAF5D7F0000:error:03000086:digital envelope routines:evp_md_init_internal:initialization error:crypto/evp/digest.c:254:
[root@590025fddcf8 /]# OPENSSL_CONF=DOES_NOT_EXIST openssl md5 /etc/rocky-release
MD5(/etc/rocky-release)= 0f942d6005c9fbc682fb92fdea8172e0
```

Reply:
Currently TiKV doesn't need the function. Just clean the md5 package here.

However, TiKV itself would use md5 when put object to s3:
https://github.com/tikv/tikv/blob/9993a63993e721105a2d52b949cd60481967173f/components/cloud/aws/src/s3.rs#L333-L338